### PR TITLE
(MODULES-4098) Sync more files

### PR DESCRIPTION
**Pull request by hunner**

### CODE REVIEW

File: config_defaults.yml [hunk 1]
```diff
@@ -123,9 +123,45 @@ Gemfile:
...
+      begin: 'latest'
+    # - name: '<name of holder>' (required)
+    #   begin: <year or 'latest'> (optional) (latest is the year when modsync is run)
+    #   end:   <year or 'latest'> (optional) (latest is the year when modsync is run)
+
 Rakefile:
   default_disabled_lint_checks:
   - 'relative'
+  - 'disable_80chars'
+  - 'disable_class_inherits_from_params_class'
```
Comment (DavidS) [inline config_defaults.yml]: Do not repeat the defaults from https://github.com/puppetlabs/puppetlabs_spec_helper/blob/master/lib/puppetlabs_spec_helper/rake_tasks.rb#L428-L434

File: moduleroot/Rakefile [hunk 2]
```diff
@@ -1,11 +1,13 @@
...
 require 'puppetlabs_spec_helper/rake_tasks'
+require 'puppet-lint/tasks/puppet-lint'
+require 'puppet_blacksmith/rake_tasks' if Bundler.rubygems.find_name('puppet-blacksmith').any?
 
+PuppetLint.configuration.fail_on_warnings = true
 <% checks = @configs['default_disabled_lint_checks'] + ( @configs['extra_disabled_lint_checks'] || [] ) -%>
 <% checks.each do |check| -%>
 PuppetLint.configuration.send('<%= check %>')
 <% end -%>
+PuppetLint.configuration.ignore_paths = ["spec/**/*.pp", "pkg/**/*.pp", "vendor/**/*.pp"]
```
Comment (DavidS) [inline moduleroot/Rakefile]: Do not override the comprehensive defaults from https://github.com/puppetlabs/puppetlabs_spec_helper/blob/master/lib/puppetlabs_spec_helper/rake_tasks.rb#L435-L442

File: moduleroot/appveyor.yml [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+<% end -%>
+notifications:
+- provider: Email
+  to:
+  - nobody@nowhere.com
+  on_build_success: false
+  on_build_failure: false
+  on_build_status_changed: false
+<% else -%>
+# Appveyor is only enabled on modules that support windows
```
Comment (DavidS) [inline moduleroot/appveyor.yml]: Having a broken appveyor file will cause issues when trying to enable appveyor as the "Only run on branches with appveyor.yml" setting will not trigger. See the results on https://github.com/puppetlabs/puppetlabs-stdlib/pull/712

**Comment by DavidS** [comment]
:tada: so excited to see this happening!

**Comment by eputnam** [comment]
rainbow pin?

File: config_defaults.yml [hunk 4]
```diff
@@ -123,9 +123,45 @@ Gemfile:
       - gem: beaker-abs
         from_env: 'BEAKER_ABS_VERSION'
         version: '~> 0.1'
+appveyor.yml:
+  appveyor_bundle_install: "bundle install --jobs 4 --retry 2 --without system_tests"
+  matrix:
+    - PUPPET_GEM_VERSION: '~> 3.0'
```
Comment (eputnam) [inline config_defaults.yml]: puppet 3?

File: moduleroot/Rakefile (hunk 2 above)
Comment (hunner) [inline moduleroot/Rakefile]: Ah right. I was wondering where these were coming from.

File: moduleroot/appveyor.yml (hunk 3 above)
Comment (hunner) [inline moduleroot/appveyor.yml]: The condition of supported OSs (linux and/or windows) is not something built-in to modulesync so it needs ERB templating to present different yaml entries. config_defaults.yaml defines the files to be managed, one of which is appveyor.yml. It is not possible to conditionalize the management of appveyor.yml, only define the defaults. And the template itself cannot afaik say "actually I should not be created" only "I should be empty".
Options:
1. appveyor.yml exists in all repos and we manually enable/disable testing (one time cost per module, imho best option)
1. appveyor.yml exists in all repos and we test on all modules (what is testing apt in appveyor supposed to do? This doesn't make sense)
1. appveyor.yml key only exists in $module/.sync.yml and we have the defaults in :global in config_defaults.yml (this is an ugly hack)
1. modulesync is expanded to have templates be conditionally managed (complex, low return, and don't want to wait for this)

Comment (hunner) [inline moduleroot/appveyor.yml]: This only appears to happen on apt and stdlib?

Comment (DavidS) [inline moduleroot/appveyor.yml]: We don't currently have the tests setup to do so, but long-term testing should cover whether facts and other native code can be plugin-synced to all (agent-)supported platforms and not break a run when none of the dependencies are installed.

For some modules, like stdlib, we should already be testing everything on windows, as it is a supported platform.

or, 5. manually set `appveyor.yml` to `unmanaged: true` in the local `.sync.yml`.

apt and stdlib are the only ones where I started playing around with appveyor.

**Pull request merged**